intake/source/derived.py

```diff
@@ -268,3 +268,68 @@ def __init__(self, columns, **kwargs):
 
     def pick_columns(self, df):
         return df[self._params["columns"]]
+
+
+class DataFramePipeline(DataFrameTransform):
+    """Apply a sequence of transformations over a DataFrame
+
+    The sequence of steps is defined as a list-of-dicts under
+    the key "steps". Each step requires a "method" key and
+    can optionally take "kwargs".
+
+    A special method called 'cols' enables column selection
+    and takes the kwarg 'columns'.
+
+    If a method cannot be found as an attribute of the output
+    of the previous step, the pipeline will attempt to import
+    the method and apply it.
+
+    Here's an example of performing a groupby, selecting two
+    columns and computing the mean.
+
+    by_origin:
+      driver: intake.source.derived.DataFramePipeline
+      args:
+        targets:
+          - auto
+        steps:
+          - method: groupby
+            kwargs:
+              by: origin
+          - method: cols
+            kwargs:
+              columns:
+                - hp
+                - mpg
+          - method: mean
+    """
+
+    input_container = "dataframe"
+    container = "dataframe"
+    required_params = ["steps"]
+
+    def __init__(self, steps, **kwargs):
+        kwargs.update(transform=self.pipeline, steps=steps, transform_kwargs={})
+        super().__init__(**kwargs)
+
+    def pipeline(self, df):
+        """Apply pipeline steps"""
+        for step in self._params["steps"]:
+            method = step["method"]
+            kwargs = step.get("kwargs", {})
+            if callable(method):
```

Comment:
It would never be a callable unless constructed dynamically in a session, right? Or did you think to use `import_name` here?

Reply:
I think that's the case. I've not practiced making a pipeline dynamically, I've found trying to do that somewhat difficult. I'll give it a try again.

Later on I have a fallback to import_name if the method doesn't look like a member method or special case.